syntax = "proto3";

message Composition {
  string type = 1;
  float percentage = 2;
}

message Orbit {
  float period = 1;
  float rotational_period = 2;
  bool tidally_locked = 3;
  float periapsis = 4;
  float eccentricity = 5;
  float inclination = 6;
  float semimajor_axis = 7;
  float axial_tilt = 8;    
}

message Parent {
  string type = 1;
  int32 relative_id = 2;
}

message Ringlike {
  string name = 1;
  string type = 2;
  float mass = 3;
  float inner_radius = 4;
  float outer_radius = 5;
}

message Planet {
  message PlanetMetadata {
    string type = 1;
    float distance = 2;
    float mass = 3;
    float gravity = 4;
    bool landable = 5;
    float radius = 6;
    int64 temperature = 7;
    float pressure = 8;
    string volcanism = 9;
    string terraforming = 10;
  }

  int64 id = 1;
  int64 system_id = 2;
  int64 relative_id = 3;
  string name = 4;
  PlanetMetadata metadata = 5;
  repeated Composition atmosphere = 6;
  repeated Composition composition = 7;
  repeated Composition materials = 8;
  repeated Ringlike rings = 9;
  repeated Ringlike belts = 10;
  repeated int64 parents = 11;
  Orbit orbit = 12;
  int64 updated = 13;
}

message Star {
  message StarMetadata {
    string type = 1;
    float distance = 2;
    string reserve_level = 3;
    string spectral_class = 4;
    float solar_masses = 5;
    float solar_radius = 6;
    string luminosity = 7;
    int64 temperature = 8;
  }

  int64 id = 1;
  int64 system_id = 2;
  int32 relative_id = 3;
  string name = 4;
  StarMetadata metadata = 5;
  repeated Ringlike rings = 6;
  repeated Ringlike belts = 7;
  repeated Parent parents = 8;
  Orbit orbit = 9;
  int64 updated = 10;
}
